move-vm-backend-common/tests/move_bytecode.rs

```diff
@@ -0,0 +1,279 @@
+//! Integration tests for the move-bytecode module.
+//!
+//! Note:
+//! These test heavily depend on Move projects within tests/assets/move-projects.
+//! Some of these tests use addresses that need to match the address in Move project files -
+//! otherwise executing scripts or publishing won't work as expected.
+
+use move_vm_backend_common::move_bytecode::verify_script_integrity_and_check_signers;
+
+/// Reads bytes from a file for the given path.
+/// Panic if the file doesn't exist.
+fn read_bytes(file_path: &str) -> Vec<u8> {
+    std::fs::read(file_path)
+        .unwrap_or_else(|e| panic!("Can't read {file_path}: {e} - make sure you run move-vm-backend/tests/assets/move-projects/smove-build-all.sh"))
+}
+
+/// Reads a precompiled Move scripts from our assets directory.
+fn read_script_bytes_from_project(project: &str, script_name: &str) -> Vec<u8> {
+    const MOVE_PROJECTS: &str = "tests/assets/move-projects";
+
+    let path =
+        format!("{MOVE_PROJECTS}/{project}/build/{project}/bytecode_scripts/{script_name}.mv");
+
+    read_bytes(&path)
+}
+
+// General tests.
+
+#[test]
+fn general_invalid_script_fails() {
+    // An invalid script like: "Bad cafe, aaaaa"!
+    let script = vec![0xBA, 0xD, 0xCA, 0xFE, 0xAA, 0xAA, 0xAA, 0xAA];
+    let result = verify_script_integrity_and_check_signers(&script);
+
+    assert!(result.is_err());
+}
+
+#[test]
+fn general_script_no_params_works() {
+    let script = read_script_bytes_from_project("signer_scripts", "no_param_at_all");
+    let expect_no_signers = verify_script_integrity_and_check_signers(&script).unwrap();
+
+    assert_eq!(expect_no_signers, 0);
+}
+
+#[test]
+fn general_script_no_signers_param_at_all_works() {
+    let script = read_script_bytes_from_project("signer_scripts", "no_signers_param_at_all");
+    let expect_no_signers = verify_script_integrity_and_check_signers(&script).unwrap();
+
+    assert_eq!(expect_no_signers, 0);
+}
+
+#[test]
+fn general_script_eight_normal_signers_works() {
+    let script = read_script_bytes_from_project("signer_scripts", "eight_normal_signers");
```

Comment:
I just realise, that I really like this style :blush: The script names are that clear enough to understand directly, what you can expect for the test case :+1: :+1: :+1: